src/pykeen/models/baseline/utils.py

```diff
@@ -60,3 +65,69 @@ def marginal_score(
     # note: we need to work with dense arrays only to comply with returning torch tensors. Otherwise, we could
     # stay sparse here, with a potential of a huge memory benefit on large datasets!
     return torch.from_numpy(scores.todense())
+
+
+def sparsify(
+    matrix: numpy.ndarray,
+    threshold: Optional[float] = None,
+) -> scipy.sparse.spmatrix:
+    """
+    Sparsify a matrix.
+
+    :param matrix: shape: (m, n)
+        the (dense) matrix
+    :param threshold:
+        the absolute threshold for sparsification
+
+    :return: shape: (m, n)
+        a sparsified matrix
+    """
+    if threshold is not None:
+        matrix = numpy.copy(matrix)
+        matrix[matrix < threshold] = 0.0
+    sparse = scipy.sparse.csr_matrix(matrix)
+    sparse.eliminate_zeros()
+    return sparse
+
+
+def entity_pair_matrix(
+    triples_factory: CoreTriplesFactory,
+    order: Tuple[TargetColumn, TargetColumn] = (COLUMN_HEAD, COLUMN_TAIL),
+) -> scipy.sparse.spmatrix:
+    """
+    Create a sparse matrix of entity-pairs for each relation.
+
+    :param triples:
+        the triples factory
+    :param order:
+        the order of entities
+
+    :return: shape: `(num_relations, num_entities ** 2)`
+        a sparse matrix, where each row contains the one-hot encoded set of entity pairs
+    """
+    # comment: we cannot use leakage.mapped_triples_to_sparse_matrices, since we need to
+    #          retain the full column count, num_entities ** 2, for mapping back to entities
+    mapped_triples = numpy.asarray(triples_factory.mapped_triples)
+    first, second = order
+    return scipy.sparse.coo_matrix(
+        (
+            numpy.ones((mapped_triples.shape[0],), dtype=int),
+            (
+                mapped_triples[:, 1],
+                triples_factory.num_entities * mapped_triples[:, first] + mapped_triples[:, second],
+            ),
+        ),
+        shape=(triples_factory.num_relations, triples_factory.num_entities**2),
+    ).tocsr()
+
+
+def get_relation_similarity(
+    triples_factory: CoreTriplesFactory,
+    to_inverse: bool = False,
+    threshold: Optional[float] = None,
+) -> scipy.sparse.csr_matrix:
+    """Get the relation similarity."""
```

Comment:
can you please document the parameters and return values and give a short explanation what the output is

Reply:
I could switch some parts to the methods from leakage detection; for the rest I extended the doc.